File: src/taylorbot-typescript/database/repositories/GuildNameRepository.ts
import { Log } from '../../tools/Logger';
import * as pgPromise from 'pg-promise';
import { Format } from '../../modules/discord/DiscordFormatter';
import { Guild } from 'discord.js';

export class GuildNameRepository {
    readonly #db: pgPromise.IDatabase<unknown>;
    constructor(db: pgPromise.IDatabase<unknown>) {
        this.#db = db;
    }

    async add(guild: Guild): Promise<void> {
        try {
            await this.#db.none(
                'INSERT INTO guilds.guild_names (guild_id, guild_name) VALUES ($[guild_id], $[guild_name]);',
                {
                    guild_id: guild.id,
                    guild_name: guild.name
                }
            );
        }
        catch (e) {
            Log.error(`Adding guild name for ${Format.guild(guild)}: ${e}`);
            throw e;
        }
    }
}

File: src/TaylorBot.Net/Core/src/TaylorBot.Net.EntityTracker.Infrastructure/GuildName/GuildNameRepository.cs
using Dapper;
using Discord;
using TaylorBot.Net.Core.Infrastructure;
using TaylorBot.Net.EntityTracker.Domain.GuildName;

namespace TaylorBot.Net.EntityTracker.Infrastructure.GuildName;

public class GuildNameRepository(PostgresConnectionFactory postgresConnectionFactory) : IGuildNameRepository
{
    public async ValueTask AddNewGuildNameAsync(IGuild guild)
    {
        await using var connection = postgresConnectionFactory.CreateConnection();

        await connection.ExecuteAsync(
            "INSERT INTO guilds.guild_names (guild_id, guild_name) VALUES (@GuildId, @GuildName);",
            new
            {
                GuildId = guild.Id.ToString(),
                GuildName = guild.Name
            }
        );
    }
}
